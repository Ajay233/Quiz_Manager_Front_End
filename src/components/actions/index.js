export const setToTopButton = (show=false) => {
  return {
    type: "SHOW_TO_TOP_BUTTON",
    payload: show
  }
}

export const setUrl = (url="") => {
  return {
    type: "SET_URL",
    payload: url
  }
}

export const enableLightMode = () => {
  return {
    type: "SET_THEME",
    payload: false
  }
}

export const enableDarkMode = () => {
  return {
    type: "SET_THEME",
    payload: true
  }
}

export const setLoaderState = (show=false, message="", label="") => {
  return {
    type: "SET_LOADER_STATE",
    payload: {
      show: show,
      message: message,
      label: label
    }
  }
}
