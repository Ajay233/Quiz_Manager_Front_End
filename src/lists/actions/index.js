import { get } from '../../axiosRequests/requests'
import { sessionExpired } from '../../utils/session'

export const getCategories = (endpoint, jwt) => {
  return (dispatch) => {
    get(endpoint, jwt).then((response) => {
      dispatch({
        type: "SET_CATEGORIES",
        payload: response.data
      })
    }).catch((error) => {
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(dispatch);
      }
    })
  }
}
