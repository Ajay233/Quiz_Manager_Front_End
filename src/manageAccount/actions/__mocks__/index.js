export const updateUser = jest.fn(() => { return { type: "SET_USER", payload: "" } })
