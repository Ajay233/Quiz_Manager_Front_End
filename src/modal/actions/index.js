export const showModal = () => {
  return {
    type: "SHOW_MODAL",
    payload: true
  }
}

export const hideModal = () => {
  return {
    type: "HIDE_MODAL",
    payload: false
  }
}
