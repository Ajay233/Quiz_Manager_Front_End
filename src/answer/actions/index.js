import { getUsingParams, post, put, del } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { hideModal } from '../../modal/actions'
import { sessionExpired } from '../../utils/session'
import history from '../../history'

// create
export const addAnswer = (data, jwt) => {
  return (dispatch) => {
    return post("answer/create", [data], jwt).then((response) => {
      dispatch({
        type: "ADD_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer created", "success", true));
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        console.log(error.response);
        dispatch(setNotification("Error - unable to create answer", "error", true));
      }
    });
  }
}

// read
export const getAnswers = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("answer/findByQuestionId", param, jwt).then((response) => {
      dispatch({
        type: "SET_ANSWERS",
        payload: response.data
      });
    }).catch((error) => {
      dispatch(setNotification("Error retrieving answers", "error", true));
    });
  }
}

// update (there was a call to setCurrentAnswer but have removed this as it's not needed)
export const updateAnswer = (body, jwt) => {
  return (dispatch) => {
    return put("answer/update", [body], jwt).then((response) => {
      dispatch({
        type: "UPDATE_ANSWER",
        payload: response.data[0]
      })
      history.push("/editQuestion");
      dispatch(setNotification("Answer updated", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(setNotification("Error - unable to update answer", "error", true));
      }
    });
  }
}

// delete
export const deleteAnswer = (config, jwt) => {
  return (dispatch) => {
    return del("answer/delete", config, jwt).then((response) => {
      dispatch(hideModal())
      dispatch({
        type: "DELETE_ANSWER",
        payload: config.data[0]
      })
      dispatch(setNotification("Answer deleted", "success", true));
    }).catch((error) => {
      console.log(error.response);
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        dispatch(hideModal())
        dispatch(setNotification("Error - Unable to delete this answer", "error", true))
      }
    });
  }
}

export const setCurrentAnswer = (answer) => {
  return {
    type: "SET_CURRENT_ANSWER",
    payload: answer
  }
}
