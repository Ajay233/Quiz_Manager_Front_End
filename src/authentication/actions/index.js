import { post } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import history from '../../history'

export const setUser = (endpoint, loginDetails) => {

  return (dispatch) => {

    post(endpoint, loginDetails).then((response) => {
      dispatch({
        type: "SET_USER_LOGGED_IN",
        payload: {
          id: response.data.user.id,
          forename: response.data.user.forename,
          surname: response.data.user.surname,
          email: response.data.user.email,
          permission: response.data.user.permission,
          verified: response.data.user.verified,
          jwt: response.data.jwt,
          loggedIn: true,
          response: response
        }
      })
      history.push("/")
      dispatch(setNotification(`Welcome back ${response.data.user.forename}`, "loginSuccess", true))
    }).catch((error) => {
      if(error.response.data === "NOT VERIFIED"){
        const verifyMsg = "Your email has not yet been veirified.  You will need to verify your email before you can log in"
        dispatch(setNotification(verifyMsg, "warning", true))
      } else {
        dispatch({
          type: "SET_USER_LOGGED_IN",
          payload: {
            error: error.response
          }
        });
        dispatch(setNotification("The username or password you entered was incorrect", "error", true));
      }
    })
  }
}

export const logOutUser = () => {
  return {
    type: "LOG_OUT_USER",
    payload: {
      forename: "",
      surname: "",
      email: "",
      permission: "",
      verified: "",
      jwt: "",
      loggedIn: false
    }
  }
}

export const setVerficationProcess = (endpoint, verificationDetails) => {

  return (dispatch) => {

    post(endpoint, verificationDetails).then((response) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "completed",
          token: "",
          error: {}
        }
      })
    }).catch((error) => {
      dispatch({
        type: "SET_VERIFY_PROCESS_STATUS",
        payload: {
          completionStatus: "not verified",
          token: verificationDetails.token,
          error: error.response
        }
      })
    })
  }
}

export const setRedirect = (status) => {
  return {
    type: "SET_REDIRECT",
    payload: {
      status: status
    }
  }
}

export const logOut = () => {
  return {
    type: "RESET_APP"
  }
}

// export const getUsers = (endpoint, header) => {
//   return {
//     type: "GET_USER",
//     payload: {
//       userId
//   }
// }
