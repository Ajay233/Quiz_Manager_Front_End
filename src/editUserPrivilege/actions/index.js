import { getUsingParams } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const setUserResults = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("users/findByEmail", param, jwt).then((response) => {
      dispatch({
        type: "SET_USER_RESULTS",
        payload: response.data
      })
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        const msg = error.response.data;
        dispatch(setNotification(msg, "error", true));
        console.log(error.response)
      }
    });
  }
}

export const clearUserResults = () => {
  return {
    type: "CLEAR_USER_RESULTS",
    payload: {}
  }
}
