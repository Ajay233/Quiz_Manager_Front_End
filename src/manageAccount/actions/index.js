import { put } from '../../axiosRequests/requests'
import { setUserData } from '../../authentication/actions'
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const updateUser = (data, jwt) => {
  const emailChangeMsg = "Profile information updated - Your email was changed so your access has been restriced until your email is verified"
  const errorMsg = "Error updating your profile data, please check the details you provided and try again";
  const emailTaken = "An account with that email already exists"
  return (dispatch) => {
    return put('users/update', data, jwt).then((response) => {
      if(response.data.jwt === undefined){
        dispatch(setUserData(response.data, jwt))
        dispatch(setNotification("Profile information updated", "success", true))
      } else {
        dispatch(setUserData(response.data.user, response.data.jwt))
        dispatch(setNotification(emailChangeMsg, "success", true))
      }
    }).catch((error) => {
      console.log(error)
      console.log(error.response)
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else if(error.response.data === emailTaken){
        dispatch(setNotification(emailTaken, "error", true))
      } else {
        dispatch(setNotification(errorMsg, "error", true))
      }
    });
  }
}
