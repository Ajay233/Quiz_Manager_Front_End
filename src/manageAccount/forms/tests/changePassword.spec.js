import React from 'react'
import { Provider } from 'react-redux'
import ChangePassword from '../changePassword'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import mockAxios from 'jest-mock-axios'

import { sessionExpired } from '../../../utils/session'
import { setNotification } from '../../../notifications/actions'

jest.mock("../../../axiosRequests/axiosUtil")
jest.mock("../../../utils/session")
jest.mock("../../../notifications/actions")

const mockStore = configureStore({})
// 44,46,47,49,50
describe("ChangePassword", () => {

  let store;
  let props;
  let wrapper;

  beforeEach(() => {
    store = mockStore({})

    props = {
        id: 1,
        email: "email",
        jwt: "Jwt"
    }

    wrapper = mount(
      <Provider store={store}>
        <ChangePassword
          userData={props}
          setNotification={setNotification}
        />
      </Provider>
    )
  })

  afterEach(() => {
  jest.clearAllMocks();
  mockAxios.reset();
  });


  it("should call axios put method when submit is clicked", () => {

    // const data = {
    //   id: 1,
    //   email: "email",
    //   password: "test",
    //   newPassword: "testChange",
    //   retypedPassword: "testChange"
    // }

    // Will need to create a separate store for these to work properly
    // wrapper.find('input').at(0).simulate('change', { target: { name: 'password', value: "test" } })
    // wrapper.find('input').at(1).simulate('change', { target: { name: 'newPassword', value: "testChange" } })
    // wrapper.find('input').at(2).simulate('change', { target: { name: 'retypedPassword', value: "testChange" } })

    const requestResponse = {
      data: "UPDATED",
      status: 200
    }
    wrapper.find('.changePasswordForm').simulate('submit')
    mockAxios.mockResponse(requestResponse)
    expect(mockAxios.put).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledTimes(1)
  })

  it("should call session expired when a 403 status error is recieved", () => {

    const errorResponse = {
      response: {
        status: 403
      }
    }

    wrapper.find('.changePasswordForm').simulate('submit')
    mockAxios.mockError(errorResponse)
    expect(mockAxios.put).toHaveBeenCalledTimes(1)
    expect(sessionExpired).toHaveBeenCalledTimes(1)

  })

  it("should call setNotification with the appropriate message for 'PASSWORD MISMATCH' error response", () => {

    const errorResponse = {
      response: {
        data: "PASSWORD MISMATCH",
        status: 400
      }
    }
    const msg = "Retyped password did not match the new password"
    wrapper.find('.changePasswordForm').simulate('submit')
    mockAxios.mockError(errorResponse)
    expect(mockAxios.put).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledWith(msg, "error", true)
  })

  it("should call setNotification with the appropriate message for 'NO MATCH' error response", () => {

    const errorResponse = {
      response: {
        data: "NO MATCH",
        status: 404
      }
    }
    const msg = "The password you provided was incorrect"
    wrapper.find('.changePasswordForm').simulate('submit')
    mockAxios.mockError(errorResponse)
    expect(mockAxios.put).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledWith(msg, "error", true)
  })

  it("should call setNotification with the appropriate message for 'PASSWORD INCORRECT' error response", () => {

    const errorResponse = {
      response: {
        data: "PASSWORD INCORRECT",
        status: 400
      }
    }
    const msg = "The password you provided was incorrect"
    wrapper.find('.changePasswordForm').simulate('submit')
    mockAxios.mockError(errorResponse)
    expect(mockAxios.put).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledWith(msg, "error", true)
  })

  it("should call setNotification with the default message fo any other error response", () => {

    const errorResponse = {
      response: {
        data: "Unrecognised error",
        status: 400
      }
    }
    const msg = "An error has occurred please ty again"
    wrapper.find('.changePasswordForm').simulate('submit')
    mockAxios.mockError(errorResponse)
    expect(mockAxios.put).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledTimes(1)
    expect(setNotification).toHaveBeenCalledWith(msg, "error", true)
  })

  it("should match the snapshot when rendered", () => {
    let component;

    component = renderer.create(
      <Provider store={store}>
        <ChangePassword userData={props} />
      </Provider>
    )

    expect(component).toMatchSnapshot();
  })

})
