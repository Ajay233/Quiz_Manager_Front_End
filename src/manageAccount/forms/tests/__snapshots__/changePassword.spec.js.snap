// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword should match the snapshot when rendered 1`] = `
<div
  id="changePassword"
>
  <form
    className="changePasswordForm"
    onSubmit={[Function]}
  >
    <div>
      <div
        id="changePasswordTitle"
      >
        Change Password
      </div>
    </div>
    <div>
      <label>
        Current Password:
      </label>
      <input
        className="inputBox"
        id="ChangePassword - password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        New Password:
      </label>
      <input
        className="inputBox"
        id="ChangePassword - newPassword"
        name="newPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
         Re-type Password:
      </label>
      <input
        className="inputBox"
        id="ChangePassword - retypedPassword"
        name="retypedPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="submit"
      id="submitChangePassword"
    >
      Submit
    </button>
  </form>
</div>
`;
